scripts/remove_invalid_social_entries.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+"""
+In preperation for OSF-8210 which will tighten social field validation, invalid entries must be removed. Pre OSF-8210 invalid social entries can be added and viewed directly through the OSF V2 API howeverthey could not be created or viewed via the OSF website.
+"""
+import sys
+import logging
+
+from website import settings
+from website.app import setup_django, init_app
+from scripts import utils as script_utils
+
+setup_django()
+from osf.models import OSFUser
+
+logger = logging.getLogger(__name__)
+
+
+def do_migration(dry):
+    logger.warn('Removing invalid social entries!')
+    if dry:
+        logger.warn('Dry_run mode')
+    for user in OSFUser.find():
+        for item in user.social.copy():
```

Comment:
Minor: Rather than copying the whole dict, you can do a set difference:

```python
for invalid_key in set(user.social.keys()) - set(settings.SOCIAL_FIELDS.keys()):
    user.social.pop(invalid_key)
```